modules/service/src/main/scala/lucuma/odb/service/ObservationWorkflowService.scala

```diff
@@ -494,37 +517,48 @@ object ObservationWorkflowService {
 
       }
 
-      private def getInfoAndValidations(
-        oids: List[Observation.Id],
-        itcClient: ItcClient[F]
-      )(using NoTransaction[F], SuperUserAccess): ResultT[F, Map[Observation.Id, (ObservationValidationInfo, ObservationValidationMap)]] =
-        services.transactionallyT:
-          for
-            infos <- obsInfos(oids, itcClient)
-            ress  <- observationValidationsImpl(infos)
-          yield (infos, ress).tupled
+      private def computeWorkflows(
+        infos: Map[Observation.Id, ObservationValidationInfo],
+        errs:  Map[Observation.Id, ObservationValidationMap],
+        execs: Map[Observation.Id, ExecutionState]
+      ): Result[Map[Observation.Id, ObservationWorkflow]] =
+        infos
+          .toList
+          .traverse: (oid, info) =>
+            val obsErrors = errs.get(oid).toList.flatMap(_.toList)
+            workflowStateAndTransitions(info, execs.get(oid), obsErrors.map(_.code))
+              .map: (s, ss) =>
+                 oid -> ObservationWorkflow(s, ss, obsErrors)
+          .map(_.toMap)
```

Comment:
This bit was separated from `getWorkflows` so that it could be shared by the `Obscalc` version.